src/dvc_data/hashfile/hash.py

```diff
@@ -190,5 +190,9 @@ def hash_file(
         assert ".dir" not in hash_info.value
         state.save(path, fs, hash_info)
 
-    meta = Meta(size=info["size"], isexec=is_exec(info.get("mode", 0)))
+    meta = Meta(
+        size=info["size"],
+        isexec=is_exec(info.get("mode", 0)),
+        version_id=info.get("version_id", info.get("VersionId")),
```

Comment:
Which one does fsspec use?

Reply:
s3fs uses `VersionId` which matches the aws rest API string. Our open adlfs PR uses `version_id` which matches the property returned by the azure blob storage python SDK (the azure rest API uses `versionid`)